src/test/scala/mesosphere/marathon/core/task/tracker/impl/TaskStatusUpdateProcessorImplTest.scala

```diff
@@ -9,118 +9,160 @@ import mesosphere.marathon.Protos.MarathonTask
-    marathonSchedulerDriverHolder.driver = Some(schedulerDriver)
+    Given("an unknown task")
+    f.taskTracker.fetchTask(update.wrapped.taskId.getValue) returns None
 
-    val notifyHealthCheckManager = new NotifyHealthCheckManagerStepImpl(healthCheckManager)
+    When("we process the updated")
+    Await.result(f.updateProcessor.publish(status), 3.seconds)
 
-    val notifyRateLimiter = new NotifyRateLimiterStepImpl(launchQueue, appRepository)
+    Then("we expect that the appropriate taskTracker methods have been called")
+    verify(f.taskTracker).fetchTask(update.wrapped.taskId.getValue)
 
-    val updateTaskTrackerStep = new UpdateTaskTrackerStepImpl(taskTracker)
+    And("the update has been acknowledged")
+    verify(f.schedulerDriver).acknowledgeStatusUpdate(status)
 
-    val postToEventStream = new PostToEventStreamStepImpl(eventBus)
+    And("that's it")
+    f.verifyNoMoreInteractions()
+  }
 
-    val notifyLaunchQueue = new NotifyLaunchQueueStepImpl(launchQueue)
+  test("an update for existing task applies the side effects of all steps") {
+    fOpt = Some(new Fixture)
+
+    val origUpdate = TaskStatusUpdateTestHelper.finished
+    val status = origUpdate.wrapped.status.mesosStatus.get.toBuilder.setTaskId(TaskIdUtil.newTaskId(appId)).build()
+    val update = origUpdate.withTaskId(status.getTaskId)
+
+    Given("a known task")
+    f.taskTracker.fetchTask(update.wrapped.taskId.getValue) returns Some(marathonTask)
+    f.taskTracker.terminated(appId, update.wrapped.taskId.getValue) returns
+      Future.successful(Some(marathonTask))
+    f.appRepository.app(appId, version) returns Future.successful(Some(app))
+    And("and a cooperative launchQueue")
+    f.launchQueue.notifyOfTaskUpdate(any) returns Future.successful(None)
+
+    When("we process the updated")
+    Await.result(f.updateProcessor.publish(status), 3.seconds)
+
+    Then("we expect that the appropriate taskTracker methods have been called")
+    verify(f.taskTracker).fetchTask(update.wrapped.taskId.getValue)
+    verify(f.taskTracker).terminated(appId, update.wrapped.taskId.getValue)
+
+    And("the healthCheckManager got informed")
+    verify(f.healthCheckManager).update(status, version)
+    And("an app scale check has been triggered")
+    f.schedulerActor.expectMsg(ScaleApp(appId))
+    And("the appRepository got queried")
+    verify(f.appRepository).app(appId, version)
+    And("the launch queue rate limiter got informed")
+    verify(f.launchQueue).addDelay(app)
+    And("the launch queue has been notified")
+    verify(f.launchQueue).notifyOfTaskUpdate(any)
+    And("the update has been acknowledged")
+    verify(f.schedulerDriver).acknowledgeStatusUpdate(status)
+
+    And("the appropriate event got published on the event stream")
+    val eventCaptor = ArgumentCaptor.forClass(classOf[MesosStatusUpdateEvent])
+    verify(f.eventBus).publish(eventCaptor.capture())
+    assert(eventCaptor.getValue != null)
```

Comment:
Do you mind using Matchers here? They're used more often that asserts in our tests, so I'd consider them conventional.